For instance consider this _VERBOSE_ output from  daemon:

```
[TRACE] : SWUPDATE running :  [read_sw_version_file] : Installed superbird: Version 1.0.0
[TRACE] : SWUPDATE running :  [listener_create] : creating socket at /tmp/swupdateprog
[TRACE] : SWUPDATE running :  [network_initializer] : Main loop Daemon
[TRACE] : SWUPDATE running :  [listener_create] : creating socket at /tmp/sockinstctrl
[TRACE] : SWUPDATE running :  [network_thread] : Incoming network request: processing...
[INFO ] : SWUPDATE started :  Software Update started !
[TRACE] : SWUPDATE running :  [extract_file_to_tmp] : Found file:
	filename sw-description
	size 2539
[TRACE] : SWUPDATE running :  [extract_file_to_tmp] : Found file:
	filename sw-description.sig
	size 64
[TRACE] : SWUPDATE running :  [get_common_fields] : Version 1.0.1
[TRACE] : SWUPDATE running :  [parse_hw_compatibility] : Accepted Hw Revision : 1.0.0
[TRACE] : SWUPDATE running :  [add_properties] : Found properties for boot.img.xdpatch:
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property xdeltasrc: /dev/source_boot
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property dst_size: 16777216
[TRACE] : SWUPDATE running :  [parse_images] : Found Image: boot.img.xdpatch in device : /dev/update_boot for handler xdelta_image
[TRACE] : SWUPDATE running :  [add_properties] : Found properties for rootfs.ext2.xdpatch:
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property xdeltasrc: /dev/source_system
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property dst_size: 1073741824
[TRACE] : SWUPDATE running :  [parse_images] : Found Image: rootfs.ext2.xdpatch in device : /dev/update_system for handler xdelta_image
[TRACE] : SWUPDATE running :  [parse_scripts] : Found Script: postinstall.sh
[TRACE] : SWUPDATE running :  [parse_scripts] : Script entry without filename field.
[TRACE] : SWUPDATE running :  [add_properties] : Found properties for :
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property sha256: 8c8f496a54e5e550662af97d088ea03a4c1785df37eb8311b6359e9cd4d51e53
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property size: 1073741824
[TRACE] : SWUPDATE running :  [parse_scripts] : Found Script:
[TRACE] : SWUPDATE running :  [parse_scripts] : Script entry without filename field.
[TRACE] : SWUPDATE running :  [add_properties] : Found properties for :
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property sha256: 0838c742ff37dba4cc62e335ff6a2dc5954efb836e9ffc982eb755c93543ca1e
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property size: 16777216
[TRACE] : SWUPDATE running :  [parse_scripts] : Found Script:
[TRACE] : SWUPDATE running :  [parse_scripts] : Script entry without filename field.
[TRACE] : SWUPDATE running :  [add_properties] : Found properties for :
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property sha256: 12d1a7b1bceff361f212a31c93dac6aa18d7a86ff63288f6a40d515d0708eeba
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property size: 1073741824
[TRACE] : SWUPDATE running :  [parse_scripts] : Found Script:
[TRACE] : SWUPDATE running :  [parse_scripts] : Script entry without filename field.
[TRACE] : SWUPDATE running :  [add_properties] : Found properties for :
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property sha256: 0b9475ba9bfc5e7ec97784bb340ae405016ff476c7baba66dd143bc373aa7f71
[TRACE] : SWUPDATE running :  [add_properties_cb] : 		Property size: 16777216
[TRACE] : SWUPDATE running :  [parse_scripts] : Found Script:
[TRACE] : SWUPDATE running :  [parse_scripts] : Found Script: preinstall.sh
[TRACE] : SWUPDATE running :  [parse_bootloader] : Bootloader var: upgrade_step = 1
[TRACE] : SWUPDATE running :  [swupdate_verify_file] : Verify signed image: Read 2539 bytes
[TRACE] : SWUPDATE running :  [swupdate_verify_file] : Verified OK
[TRACE] : SWUPDATE running :  [check_hw_compatibility] : Hardware superbird-evt Revision: 1.0.0
[TRACE] : SWUPDATE running :  [check_hw_compatibility] : Hardware compatibility verified
[TRACE] : SWUPDATE running :  [extract_files] : Found file:
	filename rootfs.ext2.xdpatch
	size 450266 required
[TRACE] : SWUPDATE running :  [extract_files] : Found file:
	filename postinstall.sh
	size 406 required
[TRACE] : SWUPDATE running :  [extract_files] : Found file:
	filename boot.img.xdpatch
	size 4583899 required
[TRACE] : SWUPDATE running :  [extract_files] : Found file:
	filename preinstall.sh
	size 1095 required
[TRACE] : SWUPDATE running :  [network_initializer] : Valid image found: copying to FLASH
[ubootenv] update_bootenv_varible name [recovery_status]: value [in_progress]
[ubootenv] Save ubootenv to /dev/env succeed!
[INFO ] : SWUPDATE running :  Installation in progress
[TRACE] : SWUPDATE running :  [extract_scripts] : No script provided for script of type readfront
[TRACE] : SWUPDATE running :  [extract_scripts] : No script provided for script of type readfront
[TRACE] : SWUPDATE running :  [extract_scripts] : No script provided for script of type readback
[TRACE] : SWUPDATE running :  [extract_scripts] : No script provided for script of type readback
swupdate preinst
[ota] current slot is b
[ota] update slot is a
[ota] target system is /dev/system_a
[ota] target boot is /dev/boot_a
[ota] target dtbo is /dev/dtbo_a
[TRACE] : SWUPDATE running :  [execute_shell_script] : Calling shell script /tmp/scripts/preinstall.sh  : return with 0
[INFO ] : SWUPDATE running :  [readfront] : Entering readfront handler
[TRACE] : SWUPDATE running :  [verify] : Property offset not found, use default 0
[ERROR] : SWUPDATE failed [0] ERROR core/cpio_utils.c : copyfile : 440 : HASH mismatch : 0b9475ba9bfc5e7ec97784bb340ae405016ff476c7baba66dd143bc373aa7f71 <--> 11d601b81846d52f6a2333d4083da8de1b1e97a2148e8294af51f5c49d3adcc6
[ERROR] : SWUPDATE failed [0] ERROR handlers/readback_handler.c : verify : 129 : readfront verification failed, status=-14
[ERROR] : SWUPDATE failed [0] ERROR corelib/installer.c : install_images : 275 : execute preinstall scripts failed
[ubootenv] update_bootenv_varible name [recovery_status]: value [failed]
[ubootenv] Save ubootenv to /dev/env succeed!
[ERROR] : SWUPDATE failed [1] Installation failed !
[TRACE] : SWUPDATE running :  [network_initializer] : Main thread sleep again !
[INFO ] : No SWUPDATE running :  Waiting for requests...
[TRACE] : SWUPDATE running :  [network_initializer] : Main loop Daemon
```

every message that matches `.* : SWUPDATE \w+ : .*` would be sent over progress IPC.

--- a/corelib/progress_thread.c
+++ b/corelib/progress_thread.c
@@ -149,7 +149,7 @@ void swupdate_progress_info(RECOVERY_STATUS status, int cause, const char *info)
 {
 	struct swupdate_progress *prbar = &progress;
 	pthread_mutex_lock(&prbar->lock);
-	snprintf(prbar->msg.info, sizeof(prbar->msg.info), "{\"%d\": %s}",
+	snprintf(prbar->msg.info, sizeof(prbar->msg.info), "{\"%d\": \"%s\"}",
 			cause, info);
 	prbar->msg.infolen = strlen(prbar->msg.info);
 	prbar->msg.status = status;

